app/controllers/api/profile_controller.rb

```diff
@@ -45,7 +45,11 @@ def resource_class
   end
 
   def resource_serializer
-    CurrentUserSerializer
+    if current_user == restricted_user
```

Comment:
Make sure we're not serializing out full users in response to restricted user actions